modules/s3-bucket/main.tf

```diff
@@ -1,53 +1,83 @@
 locals {
   enabled = module.this.enabled
 
+  aws_partition = data.aws_partition.current.partition
+
   custom_policy_account_arns = [
     for acct in var.custom_policy_account_names :
-    format("arn:%s:iam::%s:root", data.aws_partition.current.partition, module.account_map.outputs.full_account_map[acct])
+    format("arn:%s:iam::%s:root", local.aws_partition, module.account_map.outputs.full_account_map[acct])
   ]
 
-  bucket_policy = var.custom_policy_enabled ? data.aws_iam_policy_document.custom_policy[0].json : var.policy
+  bucket_policy = var.custom_policy_enabled ? data.aws_iam_policy_document.custom_policy[0].json : data.template_file.bucket_policy.rendered
 }
 
 data "aws_partition" "current" {}
 
+data "template_file" "bucket_policy" {
+  template = module.bucket_policy.json
+
+  vars = {
+    id = module.this.id
+  }
+}
+
+module "bucket_policy" {
+  source  = "cloudposse/iam-policy/aws"
+  version = "0.3.0"
+
+  iam_policy_statements = var.source_policy_documents
+
+  context = module.introspection.context
+}
+
 module "s3_bucket" {
   source  = "cloudposse/s3-bucket/aws"
-  version = "0.44.2"
-
-  abort_incomplete_multipart_upload_days = var.abort_incomplete_multipart_upload_days
-  acl                                    = var.acl
-  allow_encrypted_uploads_only           = var.allow_encrypted_uploads_only
-  allow_ssl_requests_only                = var.allow_ssl_requests_only
-  block_public_acls                      = var.block_public_acls
-  block_public_policy                    = var.block_public_policy
-  bucket_name                            = var.bucket_name
-  cors_rule_inputs                       = var.cors_rule_inputs
-  force_destroy                          = var.force_destroy
-  grants                                 = var.grants
-  ignore_public_acls                     = var.ignore_public_acls
-  kms_master_key_arn                     = var.kms_master_key_arn
-  lifecycle_rules                        = var.lifecycle_rules
-  logging                                = var.logging
-  object_lock_configuration              = var.object_lock_configuration
-  policy                                 = local.bucket_policy
-  privileged_principal_actions           = var.privileged_principal_actions
-  privileged_principal_arns              = var.privileged_principal_arns
-  restrict_public_buckets                = var.restrict_public_buckets
-  s3_replica_bucket_arn                  = var.s3_replica_bucket_arn
-  s3_replication_enabled                 = var.s3_replication_enabled
-  s3_replication_rules                   = var.s3_replication_rules
-  s3_replication_source_roles            = var.s3_replication_source_roles
-  sse_algorithm                          = var.sse_algorithm
-  transfer_acceleration_enabled          = var.transfer_acceleration_enabled
-  versioning_enabled                     = var.versioning_enabled
-  website_inputs                         = var.website_inputs
+  version = "2.0.0"
```

Comment:
Release notes need to reference [s3-bucket v2.0 upgrade instructions](https://github.com/cloudposse/terraform-aws-s3-bucket/wiki/Upgrading-to-v2.0) and in particular warn that S3-backed static websites are broken.